config/config.go

```diff
@@ -185,12 +188,17 @@ func (cfg *Config) NewNode(ctx context.Context) (host.Host, error) {
 		return nil, err
 	}
 
+	if cfg.EnableHolePunching && !cfg.Relay {
+		return nil, errors.New("cannot enable hole punching; relay is not enabled")
+	}
+
```

Comment:
I think we should enable this by default; but maybe that's for later, after we've done phase 1 testing.

Reply:
So this check gets in the way for testing with limited relays, as we will have the v1 relay disabled in this case.